models/shopify__transactions.sql

```diff
@@ -1,7 +1,22 @@
+{{
+    config(
+        materialized='incremental',
+        unique_key='transactions_unique_id',
+        incremental_strategy='merge' if target.type not in ('postgres', 'redshift', 'snowflake') else 'delete+insert',
+        cluster_by=['transaction_id']
+        ) 
+}}
+
 with transactions as (
-    select *
+    select 
+        *,
+        {{ dbt_utils.generate_surrogate_key(['source_relation', 'transaction_id'])}} as transactions_unique_id
     from {{ var('shopify_transaction') }}
 
+    {% if is_incremental() %}
+    where cast(created_timestamp as date) >= {{ shopify.shopify_lookback(from_date="max(cast(created_timestamp as date))", interval=var('lookback_window', 7), datepart='day') }}
```

Comment:
I noticed you used `created_timestamp` here as opposed to `processed_at` was there any reasoning for created over processed? I agree that `created_timestamp` is preferred. However, I want to make sure we note down why we used that here in place of another reliable timestamp. Should our standard just be `created` or `updated` timestamps for incremental models?

Reply:
Make sure to document why created-timestamp was chosen--this record would not be changed once created, but may not be processed yet.